pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -295,14 +291,10 @@ func legacyDeviceSetPVCID(deviceSetName string, setIndex int) string {
 
 // This is the new function that generates the labels
 // It includes the pvcTemplateName in it
-func deviceSetPVCID(deviceSetName, pvcTemplateName string, setIndex int) (string, error) {
+func deviceSetPVCID(deviceSetName, pvcTemplateName string, setIndex int) string {
 	cleanName := strings.Replace(pvcTemplateName, " ", "-", -1)
 	deviceSetName = strings.Replace(deviceSetName, ".", "-", -1)
-	pvcID := fmt.Sprintf("%s-%s-%d", deviceSetName, cleanName, setIndex)
-	if len(pvcID) > 62 {
-		return "", fmt.Errorf("the OSD PVC name requested is %q (length %d) is too long and must be less than 63 characters, shorten either the storageClassDeviceSet name or the storage class name", pvcID, len(pvcID))
-	}
```

Comment:
From my read of the code, this `pvcID` is used in the `generateName` field of a PersistentVolumeClaim object. If that's the case, isn't this check still valuable to users? We don't want to allow users to input names that are so long that Rook's install breaks.

Reply:
The bigger overall issue that is that even if the deviceSetName is limited to 50 chars, the pvcTemplateName can still be up to 62/63 chars. It would be better to fix this by hashing some portion of the name if it exceeds 62 chars.

With deviceSetName limited to 50 chars, and pvcTemplateName to 63, this could be up to 113 chars long just with those 2. Then 2 dashes (`-`), and an index that in theory could be 4 chars long in a ridiculous cluster. That's 119 chars.

To really fix this issue, we have to reduce these 119 chars to 62. If necessary, we can stop using the PVC ID (name) as a static value and instead use labels to `Get()` the right PVC later using labels like `pvcTemplateName=X`, `deviceSetName=Y`, `deviceSetIndex=Z`.